// Copyright 2020 Intel Corporation
//
// SPDX-License-Identifier: Apache-2.0

syntax = "proto2";
option go_package = ".;credentials";


message ccParameters
{
    optional bytes ccId = 1;
    optional bytes ccVersion = 2;
    optional int64 ccSequence = 3;
    optional bytes channelId = 4;
    optional bytes mspId = 5;
    optional bytes tlccMREnclave = 6;    
}

message ccPublicKeys
{
    optional bytes enclaveVerifyingKey = 1;
    optional bytes enclaveEncryptionKey = 2;
}

message ccPrivateKeys
{
    optional bytes enclaveSigningKey = 1;
    optional bytes enclaveDecryptionKey = 2;
}

message credentials
{
    optional bytes verb = 1;
    optional bytes enclaveAttestation = 2;
    optional bytes attestedData = 3;
    optional bytes verificationReport = 4;
}

